system-int-test/test/cmr/system_int_test/search/variable/concept_retrieval_test.clj

```diff
@@ -0,0 +1,109 @@
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.url-helper :as url]
+   [cmr.system-int-test.utils.variable-util :as variable]
+   [cmr.transmit.config :as transmit-config]
+   [cmr.umm-spec.test.expected-conversion :as expected-conversion]
+   [cmr.umm-spec.test.location-keywords-helper :as lkt]
+   [cmr.umm-spec.umm-json :as umm-json]
+   [cmr.umm-spec.umm-spec-core :as umm-spec]
+   [cmr.umm-spec.versioning :as ver]
+   [cmr.umm.umm-core :as umm]
+   [cmr.umm.umm-spatial :as umm-s]))
+
+(use-fixtures
+ :each
+ (join-fixtures
+  [(ingest/reset-fixture {"provguid1" "PROV1"
+                          "provguid2" "PROV2"
+                          "provguid3" "PROV3"})
+   variable/grant-all-variable-fixture]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Retrieve by concept-id - general test
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest retrieve-variable-by-concept-id
+  (let [var1-name "variable1"
+        var1 (variable/ingest-variable-with-attrs {:Name var1-name})
+        var2 (variable/ingest-variable-with-attrs {:Name "variable2"})
+        del-var (variable/ingest-variable-with-attrs {:Name "deleted-variable"})
+        ;; tokens
+        _ (index/wait-until-indexed)
+        del-concept (mdb/get-concept (:concept-id del-var))
+        guest-token (e/login-guest (s/context))
+        user1-token (e/login (s/context) "user1")]
+    (ingest/delete-concept del-concept (variable/token-opts user1-token))
+    (index/wait-until-indexed)
+    (testing "retrieval of a deleted variable results in a 404"
+      (let [{:keys [status errors]} (search/get-search-failure-xml-data
+                                     (search/retrieve-concept
+                                      (:concept-id del-var)
+                                      nil
+                                      {:accept mt/any
+                                       :throw-exceptions true
+                                       :headers {transmit-config/token-header user1-token}}))]
+        (is (= 404 status))
+        (is (= ["Concept with concept-id [V1200000016-PROV1] could not be found."]
+               errors))))
+    (testing "retrieval by variable concept-id returns the latest revision."
+      (let [response (search/retrieve-concept
+                       (:concept-id var1)
+                       nil
+                       {:accept mt/any
+                        :headers {transmit-config/token-header user1-token}})]
+        (is (= var1-name
+               (:Name (json/parse-string (:body response) true))))))
+    (testing "retrieval by variable concept-id and revision-id returns the specified variable"
```

Comment:
Add a new revision with a different variable_name, this way you can be sure that the test is actually doing what it claims to do.